presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -909,6 +931,34 @@ public void dropTable(String databaseName, String tableName, boolean deleteData)
         }
     }
 
+    private static void deleteTableFilesIfNeeded(HdfsContext context, HdfsEnvironment hdfsEnvironment, String databaseName, String tableName, Path tablePath)
+    {
+        try {
+            if (isS3Table(tablePath)) {
+                FileSystem fs = hdfsEnvironment.getFileSystem(context, tablePath);
+                if (fs.listFiles(tablePath, true).hasNext()) {
```

Comment:
GlueMetastore is using`fs.delete(tablePath, recursive=true)` to delete - can it be used here as well?

Reply:
https://github.com/prestosql/presto/blob/e46dd631f4162b1247ff1fa8a0974f08fe946649/presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java#L941
Same here, but only after checking if any files left after the drop table (Remember: it doesn't happen for create table with data).

Reply:
https://github.com/prestosql/presto/blob/e46dd631f4162b1247ff1fa8a0974f08fe946649/presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java#L941
Same here, but only after checking if any files left after the drop table (Remember: it doesn't happen for create table with data).

Reply:
This shouldn't be needed, since the delete call (see code in `PrestoS3FileSystem`) will do the list and delete each file separately. We shouldn't need to list separately here.